### PR TITLE
SED-3535-refactor-step-distribution-and-step-enterprise-distribution-layouts

### PR DESCRIPTION
**Pull request by cl-exense**

### CODE REVIEW

File: build_parameters.json [hunk 1]
```diff
@@ -6,8 +6,8 @@
 		{
 			"ACTION": "COMPILE",
 			"TYPE": "POM",
-			"FOLDER": "step-distribution-parent/",
-			"BEFORE": "step-os-upgrade-distribution",
+			"FOLDER": ".",
+			"BEFORE": "step-os-upgrade-distribution-new",
```
Comment (david-stephan) [inline build_parameters.json:10]: I guess we need to keep both plans for older branches, may be we can find a better name to make it more clear in the build server?

File: step-distribution-agent/pom.xml [hunk 2]
```diff
@@ -75,7 +59,7 @@
 						<configuration>
 							<rules>
 								<requireFilesSize>
-									<maxsize>15000000</maxsize>
+									<maxsize>20000000</maxsize>
```
Comment (david-stephan) [inline step-distribution-agent/pom.xml:62]: The size of the distribution should not increase with this PR, why this change?

Comment (cl-exense) [reply step-distribution-agent/pom.xml:62]: Because I'm constantly getting failing distribution builds on other branches when the dependencies get updated, and the sizes have to be constantly adjusted upwards, which can get pretty frustrating after a while. For instance here: https://github.com/exense/step-enterprise-distribution/compare/master...SED-3511-experimental-branch-without-sikulix.

This was therefore a **proactive** change because the sizes will increase anyway, and this check is really only meant to prevent the distribution from accidentally going completely out of bounds. That's also why I changed it to 20M instead of the bare minimum, as it would only be a matter of time until we reach the next unexpected build failure only related to size checks...

File: build_parameters.json (hunk 1 above)
Comment (cl-exense) [reply build_parameters.json]: Plan updated and working with both layouts now.

**Review by david-stephan — APPROVED** [review]
